doc/MiniLanguageTutorial.md

```diff
@@ -12,63 +10,59 @@ A Mini program is written as a set of source code files.  You can write separate
+The head declarations must come before all other declarations. It is an error to put a head declaration after a body declaration.
 
-Within each group (import or non-import) the order of declarations does not matter.
+Within each group the order of declarations does not matter.
 
-### Import declarations
+### Head Declarations
 
-`import` type *name*;
+`use` *path* `::` *name*;
 
-> This declares an imported type, which is assumed to be defined in another source code file.  The code in the local file can refer to the type as *name*, but it cannot know anything about the internals of the type.  The only operations that can be done on imported types are operations that are valid for any type.
+> This imports the type or function with name *name* in the mini file with virtual path *path*. The code in the local file can refer to the type as *name*.
 
-`import [impure] func` *name* ( *argname1: type1, argname2: type2, ...* ) *returntype*;
+`const` *ident* `=` *unsignedInt* `;`
 
-> This declares an imported function, which is assumed to be defined in another source code file. The code in the local file can call this function as if it were in the local file. (The linker will generate an error if this function is called in the local file but is not provided by another file that is being linked.) The syntax follows function declaration syntax (as defined below), except that an import ends with a semicolon where a local function declaration would instead have the function's code.
->
-> The optional `impure` modifier specifies that the function is impure, as defined below.
->
-> The compiler does not check whether the type signature declared here matches the type signature of the actual implementation of the function elsewhere.  If the type signatures are different, this might lead to a runtime error.
->
-> [Potential change: we could add the word unsafe to this somehow. Then we could say that every construction that can lead to an undetected type error has the word `unsafe` in its name.]
->
-> [Potential improvement: allow interfaces to be specified in separate files, and allow a code file to say that it implements or uses a particular interface. This would make it easier to maintain consistency, and easier for the compiler to check consistency..]
+> This declares a constant with name *ident* and value *unsignedInt*. This constant is only usable in the file it is defined.
 
-### Non-import declarations
+### Body declarations
 
 `type` *name* = *type*
 
 > This declares a type alias, allowing name to be used as a synonym for the specified type.  (Note that in the current syntax there is not a semicolon at the end.)
 
-var *name* : *type* ;
+`var` *name* : *type* ;
 
 > This declares a global variable. If type is an atomic type, the variable will be initialized to the zero value for that type. Otherwise the variable will be uninitialized. Reading an uninitialized variable before initializing it will cause undefined behavior.
```

Comment:
Globals are now assigned their default values, which we should probably detail in this document.

Reply:
Apologies, it seems my most recent commit wasn't pushed to remote.